assets/js/event-providers/utils.js

```diff
@@ -0,0 +1,229 @@
+ */
+
+// PII types for classification.
+export const PII_TYPE = {
+	EMAIL: 'email',
+	PHONE: 'phone',
+	NAME: 'name',
+};
+
+/**
+ * Normalizes a value for use in conversion tracking.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} value The value to normalize.
+ * @return {string} The normalized value.
+ */
+export function normalizeValue( value = '' ) {
+	return value.trim().toLowerCase();
+}
+
+/**
+ * Normalizes an email address for conversion tracking.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} email The email address to normalize.
+ * @return {string} The normalized email address.
+ */
+export function normalizeEmail( email ) {
+	const normalizedEmail = normalizeValue( email );
+
+	const atIndex = normalizedEmail.lastIndexOf( '@' );
+
+	// If there is no '@' in the email, return it as is.
+	if ( atIndex === -1 ) {
+		return normalizedEmail;
+	}
+
+	const domain = normalizedEmail.slice( atIndex + 1 );
+
+	// Check if it is a 'gmail.com' or 'googlemail.com' address.
+	if ( [ 'gmail.com', 'googlemail.com' ].includes( domain ) ) {
+		const prefix = normalizedEmail.slice( 0, atIndex );
+
+		// Remove dots from the prefix.
+		const normalizedPrefix = prefix.replace( /\./g, '' );
+
+		return `${ normalizedPrefix }@${ domain }`;
+	}
+
+	return normalizedEmail;
+}
+
+/**
+ * Normalizes a phone number for conversion tracking.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} phone The phone number to normalize.
+ * @return {string} The normalized phone number.
+ */
+export function normalizePhone( phone ) {
+	const normalizedPhone = normalizeValue( phone );
+
+	// Remove all non-numeric characters.
+	const digits = normalizedPhone.replace( /\D/g, '' );
+
+	// If the phone number starts with a '+' sign, keep it.
+	if ( normalizedPhone.startsWith( '+' ) ) {
+		return `+${ digits }`;
+	}
+
+	return digits;
+}
+
+/**
+ * Checks if a value is likely an email address.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} value The value to check.
+ * @return {boolean} True if the value is likely an email address, false otherwise.
+ */
+export function isLikelyEmail( value ) {
+	const normalizedEmail = normalizeEmail( value );
+
+	const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+	return emailPattern.test( normalizedEmail );
+}
+
+/**
+ * Checks if a value is likely a phone number.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} value The value to check.
+ * @return {boolean} True if the value is likely a phone number, false otherwise.
+ */
+export function isLikelyPhone( value ) {
+	const normalizedPhone = normalizePhone( value );
+
+	const phonePattern = /^\+?\d{7,}$/;
+
+	return phonePattern.test( normalizedPhone );
+}
+
+/**
+ * Classifies a field as PII based on its metadata.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} fieldMeta The metadata of the field to classify.
+ * @return {Object} An object containing the PII type and normalized value.
+ */
+export function classifyPII( fieldMeta ) {
+	let { type, name, value, label } = fieldMeta;
```

Comment:
Can we have a fallback empty object to prevent destructuring errors?
```suggestion
	let { type, name, value, label } = fieldMeta || {};
```

Reply:
Updated, thank you!